src/SQLStore/PropertyTableInfoFetcher.php

```diff
@@ -0,0 +1,219 @@
+		DataItem::TYPE_NUMBER     => 'smw_di_number',
+		DataItem::TYPE_BLOB       => 'smw_di_blob',
+		DataItem::TYPE_BOOLEAN    => 'smw_di_bool',
+		DataItem::TYPE_URI        => 'smw_di_uri',
+		DataItem::TYPE_TIME       => 'smw_di_time',
+		DataItem::TYPE_GEO        => 'smw_di_coords', // currently created only if Semantic Maps are installed
+		DataItem::TYPE_WIKIPAGE   => 'smw_di_wikipage',
+		//DataItem::TYPE_CONCEPT    => '', // _CONC is the only property of this type
+	);
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param array $customFixedProperties
+	 */
+	public function setCustomFixedPropertyList( array $customFixedProperties ) {
+		$this->customFixedPropertyList = $customFixedProperties;
+	}
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param array $customSpecialProperties
+	 */
+	public function setCustomSpecialPropertyList( array $customSpecialProperties ) {
+		$this->customSpecialPropertyList = $customSpecialProperties;
+	}
+
+	/**
+	 * Find the id of a property table that is suitable for storing values of
+	 * the given type. The type is specified by an SMW type id such as '_wpg'.
+	 * An empty string is returned if no matching table could be found.
+	 *
+	 * @since 1.8
+	 *
+	 * @param string $dataTypeTypeId
+	 *
+	 * @return string
+	 */
+	public function findTableIdForDataTypeTypeId( $dataTypeTypeId ) {
+		return $this->findTableIdForDataItemTypeId(
+			DataTypeRegistry::getInstance()->getDataItemId( $dataTypeTypeId )
+		);
+	}
+
+	/**
+	 * Find the id of a property table that is normally used to store
+	 * data items of the given type. The empty string is returned if
+	 * no such table exists.
+	 *
+	 * @since 1.8
+	 *
+	 * @param integer $dataItemId
+	 *
+	 * @return string
+	 */
+	public function findTableIdForDataItemTypeId( $dataItemId ) {
+
+		if ( array_key_exists( $dataItemId, $this->defaultDiTypeTableIdMap ) ) {
+			return $this->defaultDiTypeTableIdMap[$dataItemId];
+		}
+
+		return '';
+	}
+
+	/**
+	 * Retrieve the id of the property table that is to be used for storing
+	 * values for the given property object.
+	 *
+	 * @since 1.8
+	 *
+	 * @param DIProperty $property
+	 *
+	 * @return string
+	 */
+	public function findTableIdForProperty( DIProperty $property ) {
+
+		if ( $this->fixedPropertyTableIds === null ) {
+			$this->getPropertyTableDefinitions();
+		}
+
+		$propertyKey = $property->getKey();
+
+		if ( array_key_exists( $propertyKey, $this->fixedPropertyTableIds ) ) {
+			return $this->fixedPropertyTableIds[$propertyKey];
+		}
+
+		return $this->findTableIdForDataTypeTypeId( $property->findPropertyTypeID() );
+	}
+
+	/**
+	 * Return the array of predefined property table declarations, initialising
+	 * it if necessary. The result is an array of SMWSQLStore3Table objects
+	 * indexed by table ids.
+	 *
+	 * It is ensured that the keys of the returned array agree with the name of
+	 * the table that they refer to.
+	 *
+	 * @since 2.2
+	 *
+	 * @return TableDefinition[]
+	 */
+	public function getPropertyTableDefinitions() {
+
```

Comment:
You can do 

```php
if ( $this->propertyTableDefinitions === null ) {
    $this->buildCache();
}

return $this->propertyTableDefinitions;
```

Then you don't mix the caching part in with regular access, and avoid repeating the return of the field.